src/glossary.md

```diff
@@ -175,10 +175,13 @@ the hierarchy has its own collection of named entities.
 Types that can be referred to by a path directly. Specifically [enums],
 [structs], [unions], and [trait objects].
 
-### Object safe traits
+### Trait-object-safe traits
 
-[Traits] that can be used as [trait objects]. Only traits that follow specific
-[rules][object safety] are object safe.
+[Traits] that can be used in `dyn Trait` types, and allow creation of
+[trait objects]. Only traits that follow specific [rules][object safety] are
+trait-object safe.
+
+This is also called *object safety*.
```

Comment:
```suggestion
[Traits] that can be used in [trait object types][trait objects] (`dyn Trait`). Only traits that follow specific [rules][dyn-compatibility] are dyn compatible.

Formerly known as *object safety*.
```

\+ updating the second part of the reference-style MarkDown link.

TC, let me know if you agree with my suggestion.

Reply:
Looks about right.  Probably worth making a full sentence out of the last line, e.g. "These were formerly known as object safe traits".